salt-mode.el

```diff
@@ -82,6 +82,50 @@ suitable for spellchecking."
 
 (put 'salt-mode 'flyspell-mode-predicate #'salt--flyspell-predicate)
 
+(defun salt--state-module-at-point ()
+  "Get the state module at point, either pkg or pkg.installed, or return nil."
```

Comment:
I think this should return bounds and register with [https://www.emacswiki.org/emacs/ThingAtPoint](thingatpt).

This finds both state modules (e.g. "file") and state functions (e.g. "file.managed"). For the eldoc code I need these to be recognized as different things. `salt-open-doc` looks like it would benefit from these being separate too; the path requires the module and the anchor requires the function, which it re-splits itself right now.